ipa-core/src/protocol/context/dzkp_validator.rs

```diff
@@ -350,6 +363,10 @@ impl MultiplicationInputsBatch {
         let position_within_block_start = (length * id_within_batch) % 256;
         let position_within_block_end = position_within_block_start + segment.len();
 
+        if self.vec.len() <= block_id {
```

Comment:
Is there an advantage of using `&mut` pointer to the element inside the vector, or we could just allocate a block, copy the segment into it and do `vec.push`? We can take ourselves off the resize business in this case

Reply:
I don't think we can assume that the records show up in order. I could resize to one less and then push, which doesn't get rid of the resize, but might help if the common case is the records arriving in order.